SourceryRuntime/Sources/Variable.swift

```diff
@@ -51,6 +51,20 @@ public final class Variable: NSObject, SourceryModel, Typed, Annotated {
         return attributes[Attribute.Identifier.final.name] != nil
     }
 
+    /// Reference to type name where the variable is defined,
+    /// nil if defined outside of any `enum`, `struct`, `class` etc
+    public let definedInTypeName: TypeName?
+
+    /// Actual type name where the variable is defined if declaration uses typealias, otherwise just a `definedInTypeName`
+    public var actualDefinedInTypeName: TypeName? {
+        return definedInTypeName?.actualTypeName ?? definedInTypeName
```

Comment:
`actualTypeName` will be not set if type is not resolved in Composer using `resolveType` method (`TypeName` is class but references are not shared), so instead of setting `definedInType` in unique types it should be done in `resolveVariable` and other methods that resolve types to make sure all types are resolved. The same as how method's returnType is resolved explicitly. There also should be tests ensuring that definitions from type aliased extensions are properly resolved.